conf/machine/apalis-imx8x-smartracks.conf

```diff
@@ -0,0 +1,73 @@
+#@TYPE: Machine
+#@NAME: Toradex Apalis iMX8X
+#@DESCRIPTION: Toradex Apalis iMX8X powered by a i.MX 8X SoC
+#@MAINTAINER: Philippe Schenker <philippe.schenker@toradex.com>
+
+MACHINE_NAME = "Apalis-iMX8X"
+
+# for C0 silicon add mx8qxpc0 to the right of mx8qxp
+SILICON-OVERRIDES ?= "mx8qxpc0:"
+MACHINEOVERRIDES =. "mx8:mx8x:mx8qxp:${SILICON-OVERRIDES}"
+
+IMX_DEFAULT_BSP = "nxp"
+
+require conf/machine/include/imx-base.inc
+# if one wants cortexa35-crypto, set DEFAULTTUNE accordingly
+DEFAULTTUNE_mx8qxp = "aarch64"
+require conf/machine/include/tune-cortexa35.inc
+
+# Don't include kernels in standard images
+RDEPENDS_${KERNEL_PACKAGE_NAME}-base = ""
+
+LOADADDR = ""
+
+# we do not want to have getty running on tty1 as we run
+# auto-serial-console there
+USE_VT = "0"
+
+KERNEL_DEVICETREE = " \
+    freescale/imx8qxp-apalis-eval.dtb \
+"
+
+KERNEL_IMAGETYPE_aarch64 = "Image.gz"
+
+UBOOT_SUFFIX = "bin"
+UBOOT_CONFIG ??= "sd"
+UBOOT_CONFIG[sd] = "apalis-imx8x_defconfig,sdcard"
+
+UBOOT_ENTRYPOINT = "0x80280000"
+UBOOT_RD_LOADADDRESS = "0xA0000000"
+
+BOOT_SPACE = "65536"
+IMX_BOOT_SEEK = "32"
+
+IMAGE_BOOT_FILES_append = " boot.scr-${MACHINE};boot.scr"
+WKS_FILE_DEPENDS_append = " u-boot-default-script"
+PREFERRED_PROVIDER_u-boot-default-script = "u-boot-distro-boot"
+
+PREFERRED_PROVIDER_virtual/kernel = "linux-toradex"
+PREFERRED_PROVIDER_virtual/kernel_preempt-rt = "linux-toradex"
+PREFERRED_PROVIDER_virtual/bootloader = "u-boot-toradex"
+# Currently we use the nxp 2020.04 downstream for mx8 as some features are
+# not yet available in 2020.07.
+PREFERRED_VERSION_u-boot-toradex = "2020.04%"
+
+MACHINE_FIRMWARE_append = " linux-firmware-usb8997"
+MACHINE_FIRMWARE_append = " firmware-imx-vpu-imx8"
+
+IMXBOOT_TARGETS = "flash"
+
+PREFERRED_PROVIDER_imx-sc-firmware = "imx-sc-firmware-toradex"
+BOARD_TYPE = "apalis"
+
+IMAGE_CLASSES_append = " image_type_tezi"
+IMAGE_FSTYPES += "teziimg"
+
+# The imx-boot container takes care for the i.MX 8 offset, so the container
+# has to be flashed at offset 0 directly
+UBOOT_BINARY_TEZI_EMMC = "imx-boot"
+OFFSET_BOOTROM_PAYLOAD = "0"
+
+TEZI_EXTERNAL_KERNEL_DEVICETREE_BOOT = "smartracks_overlay.dtbo"
+
+TORADEX_PRODUCT_IDS = "0046 0053 0054 2600"
```

Comment:
In [image_type_tezi.bbclass](http://git.toradex.com/cgit/meta-toradex-bsp-common.git/tree/classes/image_type_tezi.bbclass), around line 286, bb.fatal is called in case no product ID is found, so while we don't really use this to write any dtb to raw nand, the product ID still has to be specified.

Reply:
Ah I see. Alright I will mark this comment as Resolved